tracer/src/Datadog.Trace.SourceGenerators/AspectsDefinitions/AspectsDefinitionsGenerator.cs

```diff
@@ -186,7 +186,7 @@ private static IReadOnlyList<string> GetAspectSources(Compilation compilation, I
 
     private static Type Resolve(AttributeData? attribute)
     {
-        Type[] aspects = new Type[] { typeof(AspectClassAttribute), typeof(AspectMethodReplaceAttribute) };
+        Type[] aspects = new Type[] { typeof(AspectClassAttribute), typeof(AspectMethodReplaceAttribute), typeof(AspectMethodInsertBeforeAttribute), typeof(AspectMethodInsertAfterAttribute), typeof(AspectCtorReplaceAttribute) };
```

Comment:
Word of warning - the aspectsdefinition generator (like a couple of our other generators) is not very optimised anymore. Consequently it's doing a lot of work on every single key press. That wasn't an issue before because there weren't any aspects, but now there are, IIRC it's going to be regenerating the files with _every single keypress_ in the IDE. And it gets worse for every additional aspect you add.

Daniel is aware of this, and was working on improving it, but he's obviously a way for now. Just something to be aware of 😬

Reply:
Good to know. Thanks!